# 16S Metagenomics Notebook
![](https://img.shields.io/badge/Module-BIOC3301-red.svg)
[![](https://img.shields.io/badge/language-R-blue.svg)](https://www.r-project.org/)
[![](https://img.shields.io/badge/cluster-Cirrus-178F8B.svg)](http://www.cirrus.ac.uk/)

Here I lay bare all my scripts used in the Illumina data processing and further downstream analyses for my work.

## Setup information
- QIIME v1.9.1 was used for the initial data processing and analyses. The (serial batch/parallel) scripts were submitted as jobs to be run on Cirrus, a High Performance Computing (HPC) system. These PBS scripts are located in the 'DataProcessingScripts' folder.

- For further downstream statistical tests, I mostly worked with [R](https://www.r-project.org/). These scripts are located in the 'StatScripts' folder.

## Outline of pre-processing workflow

insert diagram here pls
