#!/bin/bash --login
#PBS -l walltime=02:00:00
#PBS -l select=1:ncpus=24
#PBS -N a_rick_and_morty_experiment
#PBS -A d411-training

cd $PBS_O_WORKDIR

#-------------------------------------------------
# Loading modules
#-------------------------------------------------
module load miniconda/python2

#-------------------------------------------------
# Setting temporary directory
#-------------------------------------------------
mkdir -p ~/qiime_tmp
export TMPDIR=~/qiime_tmp

#-------------------------------------------------
# Loading virtualenv
#-------------------------------------------------
echo "loading virtualenv"
# NB qiime1 and not qiimel (number one not letter l)
source activate qiime1

#-------------------------------------------------
# Joining Paired End Reads
#-------------------------------------------------
time join_paired_ends.py \
-m SeqPrep \
-f Read1.fastq.gz \
-r Read2.fastq.gz \
-b Index.fastq.gz \
-o join_pe_reads \
-j 40 \
-p 75 #optional quality filter applied

#-------------------------------------------------
# Demultiplexing of Fastq sequence data
#-------------------------------------------------
time split_libraries_fastq.py \
-i /lustre/home/d411/alorax1/new/2018_02_smb/join_pe_reads/seqprep_assembled.fastq.gz \
-b /lustre/home/d411/alorax1/new/2018_02_smb/join_pe_reads/seqprep_assembled.fastq_barcodes.fastq \
-m /lustre/home/d411/alorax1/new/2018_02_smb/2018_map.tsv \
-o split_libraries \
-q19 \ #optional quality filter applied
--store_qual_scores \
--rev_comp_barcode \
--rev_comp_mapping_barcodes

#-------------------------------------------------
# Picking Operational Taxonomical Units (OTUs)
#-------------------------------------------------
time pick_open_reference_otus.py \
-i /lustre/home/d411/alorax1/new/2018_02_smb/split_libraries/seqs.fna \
-o 16S_SILVA128_otus \
-r /lustre/home/d411/alorax1/silva128/SILVA_128_QIIME_release/rep_set/rep_set_16S_only/97/97_otus_16S.fasta \
-a \
-O 12 #this is for parallelising your jobs

# Or picking closed reference OTUs
time pick_closed_reference_otus.py \
-i /lustre/home/d411/alorax1/data/2018/split_libraries_none_19/seqs.fna \
-o /lustre/home/d411/alorax1/data/2018/silva_closed_otus \
-r /lustre/home/d411/alorax1/silva128/SILVA_128_QIIME_release/rep_set/rep_set_16S_only/97/97_otus_16S.fasta \
-a \
-O 18

# Or picking de novo reference OTUs
time pick_de_novo_otus.py \
-i /lustre/home/d411/alorax1/data/2018/split_libraries_none_19/seqs.fna \
-o /lustre/home/d411/alorax1/data/2018/silva_de_novo_otus \
-a \
-O 18

#-------------------------------------------------
# Core Diversity Analyses
#-------------------------------------------------
time core_diversity_analyses.py \
--recover_from_failure \
-i /lustre/home/d411/alorax1/new/2018_02_smb/16S_SILVA128_otus/otu_table_mc2_w_tax.biom \
-o cda_silva \
-m /lustre/home/d411/alorax1/new/2018_02_smb/2018_map.tsv \
-t /lustre/home/d411/alorax1/silva128/SILVA_128_QIIME_release/trees/97/97_otus.tre \
-e 35000 \ #this number is obtained by determining sequence count where curve levels off from alpha_rarefaction.py script
-a \
-O 12

source deactivate
